platform/darwin/src/MGLStyleLayer.mm

```diff
@@ -65,6 +65,56 @@ - (float)minimumZoomLevel
     return self.rawLayer->getMinZoom();
 }
 
+- (void)setTransition:(MGLTransition)transition forKey:(NSString *)key
+{
+    SEL setPropertyTransitionSelector = [self mbx_selectorForKey:key type:@"set" hasParams:YES];
+    
+    if ([self respondsToSelector:setPropertyTransitionSelector]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+        [self performSelector:setPropertyTransitionSelector withObject:[NSValue value:&transition withObjCType:@encode(MGLTransition)] ];
+#pragma clang diagnostic pop
+    }
+}
+
+- (MGLTransition)transitionForKey:(NSString *)key
+{
+    MGLTransition transition;
+    
+    SEL getPropertyTransitionSelector = [self mbx_selectorForKey:key type:@"get" hasParams:NO];
+    
+    if ([self respondsToSelector:getPropertyTransitionSelector]) {
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+        NSValue *transitionValue = [self performSelector:getPropertyTransitionSelector];
+        [transitionValue getValue:&transition];
+#pragma clang diagnostic pop
+    }
+    
+    return transition;
+}
+
+- (SEL)mbx_selectorForKey:(NSString *)key type:(NSString *)type hasParams:(BOOL)params
+{
+    NSString *camelCaseKey;
+    NSString *parameters = params ? @":" : @"";
+    if ([key length] > 1) {
+        camelCaseKey = [NSString stringWithFormat:@"%@%@", [[key substringToIndex:1] uppercaseString], [key substringFromIndex:1]];
+    } else {
+        camelCaseKey = [key uppercaseString];
+    }
+    NSString *setPropertyTransitionString = [NSString stringWithFormat:@"mbx_%@%@Transition%@", type, camelCaseKey, parameters];
+    SEL propertyTransitionSelector = NSSelectorFromString(setPropertyTransitionString);
+    
+    return propertyTransitionSelector;
+}
+
+- (NSArray *)transitionKeys
+{
+    // This is overridden by subclasses
+    return nil;
```

Comment:
Raise an MGLAbstractClassException. Search for that string for an example to follow.